test/src/unit-iterators2.cpp

```diff
@@ -801,15 +833,15 @@ TEST_CASE("iterators 2")
                     auto it = j_object.rbegin();
                     CHECK_THROWS_AS(it[0], json::invalid_iterator);
                     CHECK_THROWS_AS(it[1], json::invalid_iterator);
-                    CHECK_THROWS_WITH(it[0], "[json.exception.invalid_iterator.209] cannot use offsets with object iterators");
-                    CHECK_THROWS_WITH(it[1], "[json.exception.invalid_iterator.209] cannot use offsets with object iterators");
+                    //CHECK_THROWS_WITH(it[0], "[json.exception.invalid_iterator.209] cannot use offsets with object iterators");
```

Comment:
The reason for two different exception message is, I changed the implementation of `operator []` in the reverse iterator to calling the `operator []` from `std::reverse_iterator` directly.

```cpp
         /// access to successor
         reference operator[](difference_type n) const
         {
-            return *(this->operator+(n));
+            return base_iterator::operator[](n);
         }
```

Therefore the behavior depends on the implementation of the `operator[]` in `std::reverse_iterator`, which varies by compiler. Most of the compiler implements it using `operator-`, but a few others uses `operator[]`.

Since you have disapproved that replacing exceptions with linear time implementation, I guess I have to revert my changes in this function to pass the test.